### PR TITLE
AnacondaRecipes sync, update to perl 5.26.2

### PR DESCRIPTION
**Pull request by mbargull**
Followup to https://github.com/conda-forge/perl-feedstock/pull/21#issuecomment-399778431

It would be nice to get `perl 5.26` into `conda-forge` since the current `conda-forge-pinning` uses that version: https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/db70a93213fd82e7deea878e9ca398d449dc47a6/recipe/conda_build_config.yaml#L364-L365

This PR is just:
- https://github.com/AnacondaRecipes/perl-feedstock/tree/cf-sync
- cherry picked https://github.com/AnacondaRecipes/perl-feedstock/commit/08aef46792efa954a6695a9411358bc688f810fc
- re-rendered with current `conda-smithy`/`conda-forge-pinning`

(cc: @conda-forge/bioconda-recipes)

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -1,20 +1,47 @@
...
+fi
+if [[ ${HOST} =~ .*linux.* ]]; then
+  _config_args+=(-Dlddlflags="-shared ${LDFLAGS}")
+# elif [[ ${HOST} =~ .*darwin.* ]]; then
+#   _config_args+=(-Dlddlflags=" -bundle -undefined dynamic_lookup ${LDFLAGS}")
+fi
+# -Dsysroot prevents Configure rummaging around in /usr and
+# linking to system libraries (like GDBM, which is GPL). An
+# alternative is to pass -Dusecrosscompile but that prevents
+# all Configure/run checks which we also do not want.
```
Comment (johanneskoester) [inline recipe/build.sh]: Looks like this is causing problems with the compiler stack on conda-forge. @jakirkham @msarahan could you help us here?

Comment (johanneskoester) [reply recipe/build.sh]: @mbargull any idea how to best proceed? Shall we try to use `-Dusecrosscompile` or is there a way to make the conda-forge compiler stack compatible with sysroot.

Comment (mbargull) [reply recipe/build.sh]: I have no idea what the implications are, TBH. But no, I don't think we can use the old compilers for that -- but I don't really know what I'm doing here :see_no_evil:.
Looking at the commit history it seems like @mingwandroid had to handle those intricacies -- Ray, you can tell us what we'll be breaking by omitting `-Dsysroot`?
If it's breaking something non-negligible, we might have to wait until cf uses newer compilers..

**Comment by bgruening** [comment]
Thanks all!

**Pull request merged**